solutions/joi-15-inheritance.mdx

```diff
@@ -0,0 +1,143 @@
+---
+id: joi-15-inheritance
+source: JOI 2015
+title: Inheritance
+author: Kevin Sheng
+---
+
+# Explanation
+
+At first glance, it seems Kruskal's would be more appropriate for this problem, so let's use that.
+
+Let's first give each child their own DSU data structure to store which cities are connected with their railroads,
+and then iterate through the railroads through decreasing order of price,
+giving each railroad to the first child that can accept it. This is effectively the same as going through
+each child one at a time and constructing a maximum spanning tree for each of them individually.
```

Comment:
Briefly explain how max spanning tree is equivalent to min span tree

Reply:
*in terms of construction

Reply:
that's... kinda obvious ngl